tests/bluetooth/bsim_bt/bsim_test_advx/src/main.c

```diff
@@ -1145,6 +1157,46 @@ static void test_scanx_main(void)
 	}
 	printk("success.\n");
 
+	printk("Waiting for Periodic Advertising Report of 0 bytes...");
+	while (!is_sync_report) {
+		k_sleep(K_MSEC(100));
+	}
+	printk("done.\n");
+
+	printk("sync_report_len = %u\n", sync_report_len);
+
+	if (sync_report_len != 0) {
+		FAIL("Incorrect Periodic Advertising Report data.");
+	}
+
+	printk("Waiting for Periodic Advertising Report of %u bytes...",
+	       sizeof(per_adv_data1));
+	sync_report_len_prev = sync_report_len;
+	while (!is_sync_report || (sync_report_len == sync_report_len_prev)) {
+		is_sync_report = false;
+		k_sleep(K_MSEC(100));
+	}
```

Comment:
Here and in other places I would prefer the pattern "sleep; if (cond) { ... }" instead of "while (cond) { ... sleep; }" to avoid hanging forever in case something does not work as expected.

Reply:
the k_sleep is added in the while loop to allow context switching in the native_posix/babblesim. If there is a failure, the babblesim simulation timeout will catch the hanging. Alternative is to use k_sem_take for each check, adding complexity to the callbacks before it can k_sem_give. But to keep the test simple and to be able to debug with printks in this loop for the multiple incoming reports, while loop seems suitable.